sdk/src/dlob/DLOBSubscriber.ts

```diff
@@ -127,6 +149,48 @@ export class DLOBSubscriber {
 			];
 		}
 
+		if (!isPerp) {
+			const isMainnet = this.env === 'mainnet-beta';
+			const config = isMainnet ? configs['mainnet-beta'] : configs['devnet'];
+			const spotMarket = (isMainnet ? (SpotMarkets['mainnet-beta']) : (SpotMarkets.devnet)).find(market => market.marketIndex === marketIndex);
+			const phoenixMarket = spotMarket.phoenixMarket;
+
+			if (opts?.includePhoenixL2) {
+				// TODO : Is websocket the right thing to use here?
+				//// TODO is it better to initialize these subscribers when we create the DLOB subscriber so that they always have data available?
+				const phoenixSubscriber = new PhoenixSubscriber({
```

Comment:
i was thinking the phoenix subscriber would be inited outside of the DLOBSubscriber and then you'd pass them into this function as `fallbackL2Generators`. The serumSubscriber and phoenixSubscriber implement `L2OrderBookGenerator`